atmo/users/adapters.py

```diff
@@ -33,3 +35,11 @@ def validate_disconnect(self, account, accounts):
             raise forms.ValidationError(
                 'At least one account needs to be connected'
             )
+
+    def populate_user(self, request, sociallogin, data):
+        user = super().populate_user(request, sociallogin, data)
+        email = user_email(user)
+        if email and '@' in email:
+            username = email.split('@')[0]
+            user_username(user, username)
+        return user
```

Comment:
Nah, this method is only about [populating the various attributes](http://django-allauth.readthedocs.io/en/latest/advanced.html#creating-and-populating-user-instances). It's used as part of the signup/login flow in allauth. [Here's the code](https://github.com/pennersr/django-allauth/blob/55e12f9ec1f83f82a2013a9a840ab71e596c0aac/allauth/socialaccount/adapter.py#L84-L112). The `user_username` helper is a generic getter/setter.